### PR TITLE
[ADD] Docstring example for digestUnmodified

### PR DESCRIPTION
**Pull request by oliveralka**
# Description

Follow up on https://github.com/OpenMS/OpenMS/pull/5396.
An additional example for digestUnmodified.

# Checklist:
- [X] Make sure that you are listed in the AUTHORS file
- [X] Add relevant changes and new features to the CHANGELOG file
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] New and existing unit tests pass locally with my changes
- [X] Updated or added python bindings for changed or new classes. (Tick if no updates were necessary.)

# How can I get additional information on failed tests during CI:
If your PR is failing you can check out 
- http://cdash.openms.de/index.php?project=OpenMS and look for your PR. If you click in the column that lists the failed tests you will get detailed error messages.

# Note:
- Once you opened a PR try to minimize the number of *pushes* to it as every push will trigger CI (automated builds and test) and is rather heavy on our infrastructure (e.g., if several pushes per day are performed).

### CODE REVIEW

File: src/pyOpenMS/pxds/ProteaseDigestion.pxd [hunk 1]
```diff
@@ -26,12 +26,28 @@ cdef extern from "<OpenMS/CHEMISTRY/ProteaseDigestion.h>" namespace "OpenMS":
...
+        #         dig.digest(bsa_aaseq, result_digest)
+        #         dig.digest(bsa_aaseq, result_digest_min_max, minlen, maxlen)
+        #         print(result_digest[4].toString()) # GLVLIAFSQYLQQCPFDEHVK
+        #         print(len(result_digest)) # 57 peptides
+        #         print(result_digest_min_max[4].toString()) # LVNELTEFAK
+        #         print(len(result_digest_min_max)) # 42 peptides
+        #
+        #         # Using digestUnmodified without the need for AASequence from the EnzymaticDigestion base class
+        #         result_digest_unmodified = []
+        #         dig.digestUnmodified(StringView(bsa_oms_string), result_digest_unmodified, minlen, maxlen)
```
Comment (timosachsenberg) [inline src/pyOpenMS/pxds/ProteaseDigestion.pxd:48]: hmm the use of StringView in python is a bit ugly.

**Comment by oliveralka** [comment]
Note: If there is an empty line somewhere in the 
```
# wrap-doc:
#    blablabla
```
this will be truncated in the python help() call. 

I am also not sure if the multiline statement is working for functions, since the docstring is showing up for single lines, but not for multi lines:

```
 |  digestUnmodified(...)
 |      Cython signature: size_t digestUnmodified(StringView sequence, libcpp_vector[StringView] & output, size_t min_length, size_t max_length)
 |  
 |  getEnzymeName(...)
 |      Cython signature: String getEnzymeName()
 |      Returns the enzyme for the digestion
```

Should actually be:
```
        Size digestUnmodified(StringView sequence,
                              libcpp_vector[ StringView ]& output,
                              Size min_length,
                              Size max_length) nogil except +
            # wrap-doc:
            #   Performs the enzymatic digestion of an unmodified sequence.
            #   -----
            #   By returning only references into the original string this is very fast.
            #   -----
            #   :param sequence: Sequence to digest
            #   :param output: Digestion products
            #   :param min_length: Minimal length of reported products
            #   :param max_length: Maximal length of reported products (0 = no restriction)
            #   :returns: Number of discarded digestion products (which are not matching length restrictions)
```

@rahul799 Did you have similar issues?

**Comment by jpfeuffer** [comment]
Did you check the actual docstring of the method instead of the summary in the class help?

**Comment by jpfeuffer** [comment]
And what if you add spaces according to the current indentation on those empty lines?

**Comment by oliveralka** [comment]
Use # instead of -

**Comment by timosachsenberg** [comment]
I think this is done, right? merge?

**Review by timosachsenberg — APPROVED** [review]

File: src/pyOpenMS/pxds/ProteaseDigestion.pxd (hunk 1 above)
Comment (jpfeuffer) [reply src/pyOpenMS/pxds/ProteaseDigestion.pxd:48]: I assume you will need a char* overload for the StringView ctor and a special conversion provider that accepts python bytes (immutable) or bytearrays (mutable) whenever StringView is encountered.

**Pull request merged**